apps/client/src/views/cuesheet/cuesheet-table/cuesheet-table-elements/cuesheetCols.tsx

```diff
@@ -1,30 +1,65 @@
 import { useCallback } from 'react';
 import { CellContext, ColumnDef } from '@tanstack/react-table';
 import { CustomFields, isOntimeEvent, OntimeEvent, OntimeRundownEntry, TimeStrategy } from 'ontime-types';
+import { millisToString, removeSeconds } from 'ontime-utils';
 
 import DelayIndicator from '../../../../common/components/delay-indicator/DelayIndicator';
-import RunningTime from '../../../../features/viewers/common/running-time/RunningTime';
+import { formatDuration } from '../../../../common/utils/time';
 import { useCuesheetOptions } from '../../cuesheet.options';
 
 import MultiLineCell from './MultiLineCell';
 import SingleLineCell from './SingleLineCell';
-import TimeInputDuration from './TimeInputDuration';
+import TimeInput from './TimeInput';
 
-import style from '../CuesheetTable.module.scss';
-
-function MakeTimer({ getValue, row: { original } }: CellContext<OntimeRundownEntry, unknown>) {
+function MakeStart({ getValue, row, table }: CellContext<OntimeRundownEntry, unknown>) {
   const { showDelayedTimes, hideTableSeconds } = useCuesheetOptions();
-  const cellValue = (getValue() as number | null) ?? 0;
-  const delayValue = (original as OntimeEvent)?.delay ?? 0;
+  const update = useCallback(
+    (newValue: string) => {
+      table.options.meta?.handleUpdateTimer(row.original.id, 'timeStart', newValue);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- we skip table.options.meta since the reference seems unstable
+    [row.original.id],
+  );
+
+  const startTime = (getValue() as number | null) ?? 0;
+  const isStartLocked = (row.original as OntimeEvent)?.linkStart === null;
+  const delayValue = (row.original as OntimeEvent)?.delay ?? 0;
+
+  let formattedTime = millisToString(startTime);
+  if (hideTableSeconds) {
+    formattedTime = removeSeconds(formattedTime);
+  }
 
   return (
-    <span className={style.time}>
-      <DelayIndicator delayValue={delayValue} />
-      <RunningTime value={cellValue} hideSeconds={hideTableSeconds} />
-      {delayValue !== 0 && showDelayedTimes && (
-        <RunningTime className={style.delayedTime} value={cellValue + delayValue} hideSeconds={hideTableSeconds} />
-      )}
-    </span>
+    <TimeInput initialValue={startTime} onSubmit={update} lockedValue={isStartLocked}>
+      {formattedTime}
+      {delayValue !== 0 && showDelayedTimes && <DelayIndicator delayValue={delayValue} />}
+    </TimeInput>
+  );
+}
+
+function MakeEnd({ getValue, row, table }: CellContext<OntimeRundownEntry, unknown>) {
+  const { hideTableSeconds } = useCuesheetOptions();
```

Comment:
would it be better to pass this as part of `table.options.meta` to try a cut down on the amount of hooks in every cell?

Reply:
yes it would